test/boot_arch_test.rb

```diff
@@ -0,0 +1,170 @@
+require_relative "test_helper"
+
+Yast.import "BootArch"
+
+describe Yast::BootArch do
+  subject { Yast::BootArch }
+
+  def stub_arch(arch)
+    Yast.import "Arch"
+
+    allow(Yast::Arch).to receive(:architecture).and_return(arch)
+  end
+
+  describe ".VgaAvailable" do
+    it "returns true if it is on x86_64 architecture" do
+      stub_arch("x86_64")
+
+      expect(subject.VgaAvailable).to eq true
+    end
+
+    it "returns true if it is on i386 architecture" do
+      stub_arch("i386")
+
+      expect(subject.VgaAvailable).to eq true
+    end
+
+    it "otherwise it returns false" do
+      stub_arch("s390x")
+
+      expect(subject.VgaAvailable).to eq false
+    end
+  end
+
+  describe ".ResumeAvailable" do
+    it "returns true if it is on x86_64 architecture" do
+      stub_arch("x86_64")
+
+      expect(subject.ResumeAvailable).to eq true
+    end
+
+    it "returns true if it is on i386 architecture" do
+      stub_arch("i386")
+
+      expect(subject.ResumeAvailable).to eq true
+    end
+
+    it "returns true if it is on s390 architecture" do
+      stub_arch("s390_64")
+
+      expect(subject.ResumeAvailable).to eq true
+    end
+
+    it "otherwise it returns false" do
+      stub_arch("ppc64")
+
+      expect(subject.ResumeAvailable).to eq false
+    end
+  end
+
+  describe ".DefaultKernelParams" do
+    context "on x86_64 or i386" do
+      before do
+        stub_arch("x86_64")
+      end
+
+      it "adds parameters from boot command line" do
+        allow(Yast::Kernel).to receive(:GetCmdLine).and_return("console=ttyS0")
+
+        expect(subject.DefaultKernelParams("/dev/sda2")).to include("console=ttyS0")
+      end
+
+      it "adds additional parameters from Product file" do
+        allow(Yast::ProductFeatures).to receive(:GetStringFeature).and_return("console=ttyS0")
+
+        expect(subject.DefaultKernelParams("/dev/sda2")).to include("console=ttyS0")
+      end
+
+      it "removes splash param from command line or product file and add it silent" do
```

Comment:
Since the removal of splash is based on a regexp, I'd appreciate having testcases for less straightforward scenarios than the parameter being at the beginning of the string.